wordpress-mobile/WordPress-iOS
# Media Async: Observe changes to media for a post

Implements #8684.

This PR adds an observer method to the media coordinator to track changes to all media for a specific post:

`addObserver(_ onUpdate: @escaping ObserverBlock, forMediaFor: AbstractPost)`

When progress is reported, all observers for the post belonging to a media item will be notified, in addition to observers for that specific media item and for all media items.

**To test:**

* Add a snippet like this to `viewWillAppear` in `AztecPostViewController`:

```
_ = mediaCoordinator.addObserver({ (media, state) in
    print("Total progress: \(self.mediaCoordinator.totalProgress(for: self.post))")
}, forMediaFor: post)
```

* Upload some media to the post, and check the observer gets called.
* Check that existing media observers aren't affected (try uploading a media item to the media library)
